src/corpus.py

```diff
@@ -0,0 +1,84 @@
+import os
+import shutil
+from gensim.corpora import Dictionary
+
+CORPUS_DIR = "corpus"
+DICT_NAME = "dict"
+
+class OnlineTextCorpus():
+    """
+    OnlineTextCorpus is a wrapper around gensim corpus data structures which is
+    expected to be updated on an event-driven basis (e.g., when new data is received).
+    The corpus is frequently checkpointed to disk to allow for data recovery.
+    TODO: Add a dense matrix format to support NLP transformations.
+    """
+    def __init__(self, dir, num_checkpoints=5):
```

Comment:
I'm not sure I understand `num_checkpoints` -- we envision this will be a user-provided arg or a computed value from the last analytics run? 

If we're thinking about a human user, it might also be nice to make `dir` and `num_checkpoints` keyword rather than positional args.

Reply:
`num_checkpoints` is a way of limiting the size of the corpus storage on disk, at the expense of losing some corpus history. It's possible that users will want to have some control over that, since it directly relates to how far you can "diff" the vocab back to.